contributed_definitions/NXregion.nxdl.xml

```diff
@@ -0,0 +1,193 @@
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition 
+	category="base" 
+	xmlns="http://definition.nexusformat.org/nxdl/3.1"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
+	name="NXregion" 
+	type="group" extends="NXobject">
+	<symbols>
+		<doc>
+			These symbols will denote how the shape of the parent group's data field,
+			
+			.. math:: (D_0, ..., D_{\mathbf{O}-1}, d_0, ..., d_{\mathbf{R}-1})
+			
+			could be split into a left set of **O** outer dimensions, :math:`\boldsymbol{D}`,
+			and a right set of **R** region dimensions, :math:`\boldsymbol{d}`,
+			where the data field has rank **O** + **R**. Note **O** :math:`>= 0`.
+		</doc>
+		<symbol name="O"><doc>Outer rank</doc></symbol>
+		<symbol name="R"><doc>Region rank</doc></symbol>
+	</symbols>
+	<doc>
+		Geometry and logical description of a region of data in a parent group. When used, it could be a child group to, say, :ref:`NXdetector`.
+
+		This can be used to describe a subset of data used to create downsampled data or to derive
+		some data from that subset.
+
+		Note, the fields for the rectangular region specifiers follow HDF5’s dataspace hyperslab parameters
+		(see https://portal.hdfgroup.org/display/HDF5/H5S_SELECT_HYPERSLAB). Note when **block** :math:`= 1`,
+		then **stride** :math:`\equiv` **step** in Python slicing.
+		
+		For example, a ROI sum of an area starting at index of [20,50] and shape [220,120] in image data::
+
+			detector:NXdetector/
+				data[60,256,512]
+				region:NXregion/
+					@type  = "rectangular"
+					parent = "data"
+					start  = [20,50]
+					count  = [220,120]
+					statistics:NXdata/
+						@signal = "sum"
+						sum[60]
+
+		the ``sum`` dataset contains the summed areas in each frame.
+		Another example, a hyperspectral image downsampled 16-fold in energy::
+
+			detector:NXdetector/
+				data[128,128,4096]
+				region:NXregion/
+					@type  = "rectangular"
+					parent = "data"
+					start  = [2]
+					count  = [20]
+					stride = [32]
+					block  = [16]
+					downsampled:NXdata/
+						@signal = "maximum"
+						@auxiliary_signals = "copy"
+						maximum[128,128,20]
+						copy[128,128,320]
+
+		the ``copy`` dataset selects 20 16-channel blocks that start 32 channels apart,
+		the ``maximum`` dataset will show maximum values in each 16-channel block
+		in every spectra.
+	</doc>
+	<attribute name="type" optional="false">
```

Comment:
The name of this attribute will **collide** with a different widespread use of the `<group type="...">` attribute (names a NeXus base class):

```bash
(bluesky_2022_2) prjemian@zap:~/.../NeXus/definitions$ git grep type= | grep .nxdl.xml | grep "<group " | wc -l
502
```

the first few examples:

```bash
applications/NXarchive.nxdl.xml:    <group type="NXentry" name="entry">
applications/NXarchive.nxdl.xml:      <group type="NXuser" name="user">
applications/NXarchive.nxdl.xml:      <group type="NXinstrument" name="instrument">
applications/NXarchive.nxdl.xml:        <group type="NXsource">
applications/NXarchive.nxdl.xml:      <group type="NXsample" name="sample">
applications/NXarpes.nxdl.xml:  <group type="NXentry">
applications/NXarpes.nxdl.xml:    <group type="NXinstrument">
applications/NXarpes.nxdl.xml:      <group type="NXsource">
applications/NXarpes.nxdl.xml:      <group type="NXmonochromator" name="monochromator">
applications/NXarpes.nxdl.xml:      <group type="NXdetector" name="analyser">
applications/NXarpes.nxdl.xml:    <group type="NXsample">
```

Reply:
So would `region_type` or `shape` be appropriate?

Reply:
+1 for `region_type`